scripts/pre_commit_linter.py

```diff
@@ -343,9 +358,51 @@ def _pre_commit_linter():
     print '\n'.join(summary_messages)
     print ''
 
+    linting_status = 0
     if any([message.startswith(_MESSAGE_TYPE_FAILED) for message in
             summary_messages]):
+        linting_status = 1
+
+    # This part is used for detecting bad patterns.
```

Comment:
Adding all this here makes the function way too long. I suggest breaking this out into a separate top-level _check_bad_patterns() function.